packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.tsx

```diff
@@ -0,0 +1,405 @@
+					payload: buildSelectedDatesEventPayload(parsedDate.startOf('day'), dayjs(this.toInputValue, DATETIME_INPUT_MASK))
+				});
+				return;
+			} else {
+				this.selectRangeDatesChange.emit({ event, payload: buildSelectedDatesEventPayload(parsedDate.startOf('day')) });
+				this.toInputFocused = true;
+			}
+		}
+
+		if (parsedDate.isValid() && parsedDateTime.isValid() && date !== this.fromInputValue) {
+			this.fromInputValue = parsedDateTime.format(DATETIME_INPUT_MASK);
+			this.displayedMonth = parsedDate;
+			if (!isEmpty(this.toInputValue)) {
+				this.selectRangeDatesChange.emit({
+					event,
+					payload: buildSelectedDatesEventPayload(parsedDateTime, dayjs(this.toInputValue, DATETIME_INPUT_MASK))
+				});
+			} else {
+				this.selectRangeDatesChange.emit({ event, payload: buildSelectedDatesEventPayload(parsedDateTime) });
+				this.toInputFocused = true;
+			}
+		}
+	};
+
+	private validateEndDate = (event: CustomEvent<string>) => {
+		const date = event.detail;
+		const parsedDate = dayjs(date, DATE_INPUT_MASK);
+		const parsedDateTime = dayjs(date, DATETIME_INPUT_MASK);
+
+		if (parsedDate.isValid() && !parsedDateTime.isValid() && date !== this.toInputValue) {
+			const parsedDateFormated = parsedDate.endOf('day').format(DATETIME_INPUT_MASK);
+			this.toInputValue = parsedDateFormated;
+			this.displayedMonth = parsedDate.subtract(1, 'month');
+			if (!isEmpty(this.fromInputValue)) {
+				if (isDateAfter(parsedDate, dayjs(this.fromInputValue, DATE_INPUT_MASK))) {
+					this.selectRangeDatesChange.emit({
+						event,
+						payload: buildSelectedDatesEventPayload(dayjs(this.fromInputValue, DATETIME_INPUT_MASK), parsedDate.endOf('day'))
+					});
+				} else {
+					this.toInputValue = undefined;
+					return;
+				}
+			} else {
+				this.selectRangeDatesChange.emit({ event, payload: buildSelectedDatesEventPayload(parsedDate.endOf('day')) });
+				this.fromInputFocused = true;
+			}
+		}
+
+		if (parsedDate.isValid() && parsedDateTime.isValid() && date !== this.toInputValue) {
+			this.toInputValue = parsedDateTime.format(DATETIME_INPUT_MASK);
+			this.displayedMonth = parsedDate.subtract(1, 'month');
+			if (!isEmpty(this.fromInputValue)) {
+				if (isDateTimeAfter(parsedDateTime, dayjs(this.fromInputValue, DATETIME_INPUT_MASK))) {
+					this.selectRangeDatesChange.emit({
+						event,
+						payload: buildSelectedDatesEventPayload(dayjs(this.fromInputValue, DATETIME_INPUT_MASK), parsedDateTime)
+					});
+				} else {
+					this.toInputValue = undefined;
+				}
+			} else {
+				this.selectRangeDatesChange.emit({ event, payload: buildSelectedDatesEventPayload(parsedDateTime) });
+				this.fromInputFocused = true;
+			}
+		}
+	};
+
+	private onClickDate = ({ detail: { event, payload: date } }: CustomEvent<IClickDateEvent>): void => {
+		const clickedDate = fromISO(date).startOf('day');
+		const [selectedStartDate, selectedEndDate] = this.selectedRangeDates;
+		const inputDate = newDate(date);
+
+		if (!selectedStartDate) {
```

Comment:
this is an enormous function, try to split this for an util. That function can return the values that you need to assign.